packages/mui-material/src/IconButton/IconButton.test.js

```diff
@@ -34,13 +34,18 @@ describe('<IconButton />', () => {
     expect(container.querySelector('.touch-ripple')).not.to.equal(null);
   });
 
-  it('can disable the ripple', () => {
-    const { container } = render(
-      <IconButton disableRipple TouchRippleProps={{ className: 'touch-ripple' }}>
-        book
-      </IconButton>,
-    );
-    expect(container.querySelector('.touch-ripple')).to.equal(null);
+  ['default', 'primary'].forEach((color) => {
+    it(`can disable the ripple and hover effect for color ${color}`, () => {
+      const { container, getByRole } = render(
+        <IconButton disableRipple color={color} TouchRippleProps={{ className: 'touch-ripple' }}>
+          book
+        </IconButton>,
+      );
+      expect(container.querySelector('.touch-ripple')).to.equal(null);
+      expect(getComputedStyle(getByRole('button'), ':hover').backgroundColor).to.equal(
```

Comment:
This `:hover` argument is not valid for the `getComputedStyle` API. The second parameter expects a pseudo-element, not a pseudo-class, see https://stackoverflow.com/questions/11495535/why-doesnt-getcomputedstyle-work-with-pseudo-classes-like-hover.

It seems that we can't test this. I'm removing the test in: https://github.com/mui/material-ui/pull/34764.